spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreResultsIterable.java

```diff
@@ -50,4 +61,20 @@ public Cursor getCursor() {
 	public Iterable<T> getIterable() {
 		return this.iterable;
 	}
+
+	public Optional<Boolean> getHasNextPage() {
+		return hasNextPage.get();
+	}
+
+	public void setHasNextPageQuery(Supplier<Boolean> query) {
```

Comment:
I don't think a `public` setter makes sense here. Can you just provide all the necessary objects to determine `hasNextPage` into the constructor?